specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3601,6 +3609,283 @@
+        },
+        "domainAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The domain admin credentials."
+        }
+      }
+    },
+    "KeyvaultCredentialProperties": {
+      "description": "Credentials kept in the keyvault.",
+      "type": "object",
+      "properties": {
+        "userName": {
+          "type": "string",
+          "description": "The user name to the account."
+        },
+        "passwordKeyVaultResourceId": {
+          "type": "string",
+          "description": "The keyvault resource id to the keyvault secrets."
+        },
+        "passwordSecretName": {
+          "type": "string",
+          "description": "The keyvault secret name the password is stored in."
+        }
+      }
+    },
+    "SessionHostComponentUpdateConfigurationProperties": {
+      "description": "The session host configuration for updating agent, monitoring agent, and stack component.",
+      "type": "object",
+      "properties": {
+        "maintenanceType": {
+          "type": "string",
+          "enum": [
+            "Default",
+            "TenantAdminControlled"
+          ],
+          "x-ms-enum": {
+            "name": "SessionHostComponentUpdateType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Default",
+                "description": "Agent and other agent side components are delivery schedule is controlled by WVD Infra."
+              },
+              {
+                "value": "TenantAdminControlled",
+                "description": "TenantAdmin have opted in for Scheduled Agent Update feature."
+              }
+            ]
+          }
+        },
+        "useLocalTime": {
+          "type": "boolean",
+          "description": "Whether to use localTime of the virtual machine."
+        },
+        "maintenanceWindowTimeZone": {
+          "type": "string",
+          "description": "Time zone for maintenance as defined in https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo.findsystemtimezonebyid?view=net-5.0. Must be set if useLocalTime is true."
+        },
+        "primaryWindow": {
+          "$ref": "#/definitions/MaintenanceWindowProperties",
+          "description": "Primary Window of the maintenance. Maintenance windows are 2 hours long. We try to push agent update in this window first."
+        },
+        "secondaryWindows": {
+          "type": "array",
+          "description": "List of secondary maintenance windows. Maintenance windows are 2 hours long. We try to exercise this only when the primary window update fails.",
+          "items": {
+            "$ref": "#/definitions/MaintenanceWindowProperties"
+          }
+        }
+      }
+    },
+    "MaintenanceWindowProperties": {
+      "description": "Maintenance window starting hour and day of week.",
+      "type": "object",
+      "properties": {
+        "hour": {
```

Comment:
For learning, how do we state a property is required? and for integer type, it is possible to state min and max value?

Reply:
ex:      "required": [
        "hostPoolArmPath",
        "applicationGroupType"
      ],
      
min max I don't think is in a swagger spec for definitions

Reply:
I found this link, https://swagger.io/docs/specification/data-models/data-types/#range.  Would you please help me if this is supported or share the link of schema definition?  thanks.